libminifi/include/utils/file/FileUtils.h

```diff
@@ -640,10 +637,10 @@ class FileUtils {
     std::vector<char> buf(1024U);
     while (true) {
       ssize_t ret = readlink("/proc/self/exe", buf.data(), buf.size());
-      if (ret == -1) {
+      if (ret < 0) {
```

Comment:
`readlink`'s contract is pretty clear from the POSIX specification (https://pubs.opengroup.org/onlinepubs/9699919799/): it will return -1 on error, not negative on error. This is fine the way it is.

Reply:
I know, but I feel safer with a `< 0` check before making a narrowing cast (to unsigned) below. (To avoid unsigned vs signed comparison warnings.)

Reply:
It doesn't hurt, I just didn't understand the purpose, but I can live with it.